src/track/beatgrid.cpp

```diff
@@ -311,6 +342,12 @@ double BeatGrid::getBpmAroundPosition(double curSample, int n) const {
     return bpm();
 }
 
+double BeatGrid::calculateFirstPhraseSample(double phraseSample) const {
+	return ((int)MathUtilities::round((phraseSample - firstBeatSample()) / m_dBeatLength) %
```

Comment:
Please format your changes with git-clang-format according to our [Style Guide](https://mixxx.org/wiki/doku.php/coding_guidelines) to fix indentations and to remove any tab characters.